kernel/src/hil/radio.rs

```diff
@@ -111,6 +111,7 @@ pub trait RadioData<'a> {
     fn set_transmit_client(&self, client: &'a dyn TxClient);
     fn set_receive_client(&self, client: &'a dyn RxClient, receive_buffer: &'static mut [u8]);
     fn set_receive_buffer(&self, receive_buffer: &'static mut [u8]);
+    fn set_ack_buffer(&self, ack_buffer: &'static mut [u8]);
```

Comment:
The most recent commit makes an attempt to resolve these changes. Since the nrf52_components is altered, it will require updating additional function signatures across other board families using the nrf52_components/peripherals (this is the reason for the PR checks failing). Once I get feedback on these changes, I will update the other board families.